src/aspire/utils/blk_diag_matrix.py

```diff
@@ -0,0 +1,936 @@
+        """
+
+        A = BlkDiagMatrix(blk_partition, dtype=dtype)
+
+        for i, blk_sz in enumerate(blk_partition):
+            A[i] = np.ones(blk_sz, dtype=dtype)
+        return A
+
+    @staticmethod
+    def eye(blk_partition, dtype=np.float64):
+        """
+        Build a BlkDiagMatrix eye (identity) matrix
+
+        :param blk_partition: The matrix block partition in the form of
+        a K-element list storing all shapes of K diagonal matrix blocks,
+        where `blk_partition[i]` corresponds to the shape (number of rows
+        and columns) of the `i` diagonal matrix block.
+        :param dtype: The data type of the diagonal matrix blocks.
+        :return: A BlkDiagMatrix instance consisting of `K` eye (identity)
+        blocks.
+        """
+
+        A = BlkDiagMatrix(blk_partition, dtype=dtype)
+
+        for i, blk_sz in enumerate(blk_partition):
+            rows, cols = blk_sz
+            A[i] = np.eye(N=rows, M=cols, dtype=dtype)
+        return A
+
+    @staticmethod
+    def eye_like(A, dtype=None):
+        """
+        Build a BlkDiagMatrix eye (identity) matrix with the partition
+        structure of BlkDiagMatrix A.  Defaults to dtype of A.
+
+        :param A: BlkDiagMatrix instance.
+        :param dtype: Optional, data type of the new diagonal matrix blocks.
+        :return: BlkDiagMatrix instance consisting of `K` eye (identity)
+        blocks.
+        """
+
+        if dtype is None:
+            dtype = A.dtype
+
+        return BlkDiagMatrix.eye(A.partition, dtype=dtype)
+
+    @staticmethod
+    def zeros_like(A, dtype=None):
+        """
+        Build a BlkDiagMatrix zeros matrix with the partition
+        structure of BlkDiagMatrix A.  Defaults to dtype of A.
+
+        :param A: BlkDiagMatrix instance.
+        :param dtype: Optional, data type of the new diagonal matrix blocks.
+        :return: BlkDiagMatrix instance consisting of `K` zeros blocks.
+        """
+
+        if dtype is None:
+            dtype = A.dtype
+
+        return BlkDiagMatrix.zeros(A.partition, dtype=dtype)
+
+    @property
+    def partition(self):
+        """
+        Return the partitions (block sizes) of this BlkDiagMatrix
+
+        :return: The matrix block partition in the form of a
+        K-element list storing all shapes of K diagonal matrix blocks,
+        where `partition[i]` corresponds to the shape (number of rows and
+        columns) of the `i` diagonal matrix block.
+        """
+
+        if self._cached_blk_sizes is None:
+            blk_sizes = np.empty((self.nblocks, 2), dtype=np.int)
+            for i, blk in enumerate(self.data):
+                blk_sizes[i] = np.shape(blk)
+            self._cached_blk_sizes = blk_sizes
+        return self._cached_blk_sizes
+
+    def check_size_compatible(self, other):
+        """
+        Sanity check two BlkDiagMatrix instances are compatible in size.
+
+        :para other: The BlkDiagMatrix to compare with self.
```

Comment:
I'm not familiar enough with the docstring syntax, but does this not have to be `param` (also below)?